crates/flox/src/commands/environment.rs

```diff
@@ -203,9 +203,8 @@ impl EnvironmentCommands {
                     .catalog(&flox.nix(Default::default()), &flox.system)
                     .await
                     .context("Could not get catalog")?;
```

Comment:
We could consider writing this to catalog.json to avoid having to re-lock if e.g. someone does `flox list` twice

Reply:
- I'm fairly confident that we'll want to keep the lock whenever we generate it even with changes to `mkFloxEnv`
- catalog is evaluated with `--impure` so the eval cache is not used
- `flox list` and `flox build` evaluate different attributes, so even if we do use the eval cache at some point, we'd still manually need to ensure they reuse the same locking work

It doesn't have to block this PR, but I think it's something we should likely expect to do

Reply:
Won't list eventually use the locked manifest anyway?
If yes i guess we shouldnt block on this for the current pr..

Reply:
Right, this is a "simplistic" `flox list`. We're literally just printing bare package names at this point so let's do real work once we have real data to work with.